client/src/worldgen.rs

```diff
@@ -124,15 +124,34 @@ pub fn voxels(graph: &DualGraph, node: NodeId, chunk: Vertex) -> VoxelData {
                 let center = voxel_center(coords);
                 let cube_coords = center * 0.5;
 
-                let (voxel_mat, elevation_boost) = match trilerp(&enviros.rainfalls, cube_coords) {
-                    r if r > 2.0 => (Material::Dirt, 1.0),
-                    r if r < 1.0 => (Material::Stone, -1.0),
-                    _ => (Material::Sand, -0.5),
-                };
+                let elev = trilerp(&enviros.max_elevations, cube_coords);
+                let rain = trilerp(&enviros.rainfalls, cube_coords);
+                let temp = trilerp(&enviros.temperatures, cube_coords);
 
-                // maximum max_elevation for this voxel according to the max_elevations
-                // of the incident nodes that dictate the content of this chunk
-                let max_e = trilerp(&enviros.max_elevations, cube_coords) - elevation_boost;
+                let mut voxel_mat = Material::Dirt;
+                let max_e = elev;
+
+                if rain > 2.5 {
+                    voxel_mat = Material::Leaves;
+                }
+                if rain < -0.5 {
+                    voxel_mat = Material::Stone;
+                }
+
+                //peaks should roughly tend to be snow-covered, and valleys should roughly be watery.
+                let temp_mod = (temp + 0.5_f64).rem_euclid(7_f64);
+                if temp_mod <= 1_f64 {
+                    voxel_mat = Material::Snow;
+                }
+
+                if (temp_mod >= 3_f64) && (temp_mod <= 4_f64) {
+                    voxel_mat = Material::Water;
+                }
```

Comment:
Why is this based on temperature? Surely this will lead to physically impossible configurations?

Could we get good results from just replacing void with water when below a certain elevation instead?

Reply:
The water is there to demonstrate the place where a valley theoretically should be based on the temperature. I also didn't want a minimum water level in this case so the terrain features of the lower areas could be seen.

Reply:
Perhaps some material other than water would make more sense then? I'm also assuming that "temperature" means "slope" here, which, if so, should certainly be reflected in the code.